### PR TITLE
Improve edit box

### PR DESCRIPTION
**Pull request by zilongshanren**
fixed https://github.com/cocos2d/cocos2d-x/issues/14827

### CODE REVIEW

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBox.java [hunk 1]
```diff
@@ -249,7 +249,7 @@ public void setInputFlag(int inputFlag) {
                 this.mInputFlagConstraints = InputType.TYPE_TEXT_FLAG_CAP_CHARACTERS;
                 break;
             case kEditBoxInputFlagLowercaseAllCharacters:
-                this.mInputFlagConstraints = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS;
+                this.mInputFlagConstraints = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_NORMAL;
```
Comment (zilongshanren) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBox.java:252]: fix all lower case multiline editbox

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBoxHelper.java [hunk 2]
```diff
@@ -195,7 +195,6 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
                             //if editbox doesn't support multiline, just hide the keyboard
                             if ((editBox.getInputType() & InputType.TYPE_TEXT_FLAG_MULTI_LINE) != InputType.TYPE_TEXT_FLAG_MULTI_LINE) {
                                 Cocos2dxEditBoxHelper.closeKeyboardOnUiThread(index);
-                                mCocos2dxActivity.getGLSurfaceView().requestFocus();
```
Comment (zilongshanren) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxEditBoxHelper.java:198]: I move the focus change function into  `closeKeyboardOnUiThread` method

File: cocos/ui/UIEditBox/UIEditBoxImpl-common.cpp [hunk 3]
```diff
@@ -288,17 +291,7 @@ void EditBoxImplCommon::onEndEditing(const std::string& text)
 {
     this->setNativeVisible(false);
     
-    if(text.size() == 0)
```
Comment (zilongshanren) [inline cocos/ui/UIEditBox/UIEditBoxImpl-common.cpp:291]: it's duplicated with `refreshInactiveText`

**Comment by xpol** [comment]
I see you have make `EditBox::setVisible()` just change visibility.
Keyboard are controlled by `openKeyboard()/closeKeyboard()`.

These API are do what the supposed to. 
should fixes #14827 .

@zilongshanren Thank you very much!

**Comment by zilongshanren** [comment]
[ci rebuild]

**Pull request merged**